app/models/user.rb

```diff
@@ -0,0 +1,30 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+  
+  has n, :peeps
+
+  property :id, Serial
+  property :name, String
+  property :username, String, required: true, unique: true
+  property :email, String, required: true, unique: true
+  property :password_digest, Text
+
+  validates_confirmation_of :password
+  validates_format_of :email, as: :email_address
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(username, password)
+    user = first(username: username)
+    !!user && BCrypt::Password.new(user.password_digest) == password ?
```

Comment:
Style/DoubleNegation: Avoid the use of double negation (`!!`). (https://github.com/bbatsov/ruby-style-guide#no-bang-bang)<br>Style/MultilineTernaryOperator: Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead. (https://github.com/bbatsov/ruby-style-guide#no-multiline-ternary)